### PR TITLE
DRAFT: arrow_cast function as UDF

### LINKED ISSUE #9143: Issue using `arrow_cast` in ORDER BY expressions
### Describe the bug

originally opened in https://github.com/GlareDB/glaredb/issues/2597

It seems that `arrow_cast`'s special handling makes it so that it errors out when using it in an ORDER BY expr

### To Reproduce

```sql
statement ok
create table ts (a int);

statement ok
insert into ts values (915148800);

# this errors
statement ok 
SELECT
  DATE_TRUNC('minute', arrow_cast(a, 'Timestamp(Second, None)')) AS minute
FROM
  ts
ORDER BY
  DATE_TRUNC('minute', arrow_cast(a, 'Timestamp(Second, None)'));
```

### Expected behavior

It should be functionally equivalent in this example to using `to_timestamp_seconds`

```sql
SELECT
  DATE_TRUNC('minute', to_timestamp_seconds(a)) AS minute
FROM
  ts
ORDER BY
  DATE_TRUNC('minute', to_timestamp_seconds(a));
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by brayanjuls**
## Which issue does this PR close?

Closes #9143
Closes https://github.com/apache/arrow-datafusion/issues/9287

## Rationale for this change

Migration of core functions to UDF.

## What changes are included in this PR?

arrow_cast function migration.

## Are these changes tested?

Yes.

## Are there any user-facing changes?

No.

### CODE REVIEW

File: datafusion/functions/src/core/arrow_cast.rs [hunk 1]
```diff
@@ -19,16 +19,50 @@
...
-    plan_datafusion_err, DFSchema, DataFusionError, Result, ScalarValue,
-};
+use datafusion_common::{plan_datafusion_err, DataFusionError, Result, ScalarValue, internal_err};
 
-use datafusion_common::plan_err;
-use datafusion_expr::{Expr, ExprSchemable};
+use datafusion_expr::{ColumnarValue, ScalarUDFImpl, Signature, Volatility};
 
-pub const ARROW_CAST_NAME: &str = "arrow_cast";
+#[derive(Debug)]
```
Comment (alamb) [inline datafusion/functions/src/core/arrow_cast.rs:30]: this looks just about perfect. Thank you @brayanjuls 

I plan to get https://github.com/apache/arrow-datafusion/pull/8985 merged shortly and then this PR should be possible. 

🙏

**Comment by brayanjuls** [comment]
I messed the merge of the latest changes and somehow now they all appear in this PR. I will try to fix it, otherwise I will open a new PR.

**Comment by alamb** [comment]
Thanks @brayanjuls  -- I also filed https://github.com/apache/arrow-datafusion/issues/9287 to track this issue. I thought of a potential issue we may hit as well which I will link to https://github.com/apache/arrow-datafusion/issues/9287 as well

**Comment by brayanjuls** [comment]
@alamb I am closing this PR as I opened #9298 to fix the merging issues I have here.

**Pull request closed without merging**